hack/jenkins/metadata-cache/metadata-cache-control.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+fail() {
+  echo "$@"
+  exit 1
+}
+
+usage() {
+  fail "Usage: $(basename $0) <on|off|kill>"
+}
+
+if [[ -z "$1" ]]; then
+  usage
+fi
+
+MGI='metadata.google.internal'
+TARGET="/etc/hosts"
+
+grep "$MGI" $TARGET || fail "$MGI missing from $TARGET"
+
+case $1 in
+  on)
+    sed -i -e 's/^#\([0-9\.]\+\s\+metadata\.google\.internal\)\(\s\+.*\)*$/\1\2/g' $TARGET
+    ;;
+  off)
+    sed -i -e 's/^\([0-9\.]\+\s\+metadata\.google\.internal\)\(\s\+.*\)*$/#\1\2/g' $TARGET
+    ;;
+  kill)
+    pids=$(ps ax | grep 'python metadata-cache.py' | grep -v grep | cut -d ' ' -f 1)
```

Comment:
this works, but `pgrep` is also a neat utility that does the same thing. `pkill` actually does everything you need.